### PR TITLE
Update langname and regionname file to CLDR 36 

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [ ] `ReleaseNotes` has added or is not needed.
* [ ] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Updated `langname.jf` and `regionname.jf` file to CLDR 36.
* Updated `genlangreg.js.js` tool code file to include `XX` region value automatically.
* Added newly added language to `js/build.xml` for testing
* Added testcase regarding newly added or modified data.
* Regarding removed langname.jf files, I think that language identifiers are from `ISO 639-3` which are not in CLDR. I haven't made a decision if iLib needs to keep the data inside or not

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/data/locale/afa/langname.jf [hunk 1]
```diff
@@ -1,4 +0,0 @@
-{
-    "generated": true,
```
Comment (ehoogerbeets) [inline js/data/locale/afa/langname.jf]: Almost all of these deleted languages are actually collective language groups. That is, a name for a group of similar languages. It is okay to delete these.

File: js/data/locale/und/MK/regionname.jf [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
     "generated": true,
-    "region.name": "Macedonia"
+    "region.name": "North Macedonia"
```
[the remark this answers is not in the document]
Comment (gouniLee) [reply js/data/locale/und/MK/regionname.jf]: I'll check it

File: js/data/locale/afa/langname.jf (hunk 1 above)
Comment (gouniLee) [reply js/data/locale/afa/langname.jf]: Thank you for the confirmation!

File: tools/cldr/genlangreg.js [hunk 3]
```diff
@@ -28,32 +28,32 @@ var Locale = common.Locale;
...
 
 function usage() {
-    console.log("Usage: genlangreg [-h] [ toDir ]\n" +
+    console.log("Usage: genlangreg [-h] [ locale_data_dir ]\n" +
         "Generate the langname.jf and regionname.jf files for each locale.\n" +
         "-h or --help\n" +
         "  this help\n" +
-        "toDir\n" +
-    "  directory where generated files will be located\n");
+        "localeDirName\n" +
```
Comment (ehoogerbeets) [inline tools/cldr/genlangreg.js]: that should be "locale_data_dir" so that it matches what is in the usage above

**Pull request merged**